### ISSUE TITLE
Where can I find the code about changing learning rate?

### ISSUE THREAD

**Issue by Taruguchi** [espnet/espnet#4291]
I investigating how the results change as the learning rate change.
I refer to the learning of `vctk_noisy`.
I understand that there is the initial value `optim_conf lr:1.0-e03` in `train.yaml`.
Where is the code of changing the learning rate during training?

**Comment by Emrys365** [comment]
It depends on the `scheduler` specified in the config file.
For example, in [egs2/vctk_noisy/enh1/conf/tuning/train_enh_conv_tasnet.yaml](https://github.com/espnet/espnet/blob/master/egs2/vctk_noisy/enh1/conf/tuning/train_enh_conv_tasnet.yaml), the scheduler is configured as follows:
```yaml
scheduler: reducelronplateau
scheduler_conf:
    mode: min
    factor: 0.5
    patience: 3
```

It means the learning rate scheduler is `ReduceLROnPlateau`, by referring to the supported schedulers defined in
[espnet2/tasks/abs_task.py](https://github.com/espnet/espnet/blob/master/espnet2/tasks/abs_task.py#L145-L157):
```python
scheduler_classes = dict(
    ReduceLROnPlateau=torch.optim.lr_scheduler.ReduceLROnPlateau,
    lambdalr=torch.optim.lr_scheduler.LambdaLR,
    steplr=torch.optim.lr_scheduler.StepLR,
    multisteplr=torch.optim.lr_scheduler.MultiStepLR,
    exponentiallr=torch.optim.lr_scheduler.ExponentialLR,
    CosineAnnealingLR=torch.optim.lr_scheduler.CosineAnnealingLR,
    noamlr=NoamLR,
    warmuplr=WarmupLR,
    cycliclr=torch.optim.lr_scheduler.CyclicLR,
    onecyclelr=torch.optim.lr_scheduler.OneCycleLR,
    CosineAnnealingWarmRestarts=torch.optim.lr_scheduler.CosineAnnealingWarmRestarts,
)
```

Then you can just look into the original definition of the corresponding scheduler to understand how it works.
> In this case, `ReduceLROnPlateau` is a native LR scheduler defined in PyTorch.
